File: partiql-ast/src/main/java/org/partiql/ast/v1/Constraint.java
package org.partiql.ast.v1;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 * TODO docs, equals, hashcode
 */
public class Constraint extends AstNode {
    @Nullable
    public String name;

    @NotNull
    public ConstraintType constraintType;

    public Constraint(@Nullable String name, @NotNull ConstraintType constraintType) {
        this.name = name;
        this.constraintType = constraintType;
    }

    @NotNull
    @Override
    public Collection<AstNode> children() {
        List<AstNode> kids = new ArrayList<>();
        kids.add(constraintType);
        return kids;
    }

    @Override
    public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
        return visitor.visitConstraint(this, ctx);
    }
}

File: partiql-ast/src/main/kotlin/org/partiql/ast/v1/Constraint.kt
package org.partiql.ast.v1

/**
 * TODO docs, equals, hashcode
 */
public class Constraint(
    @JvmField
    public var name: String?,
    @JvmField
    public var constraintBody: ConstraintBody,
) : AstNode() {
    public override fun children(): Collection<AstNode> {
        val kids = mutableListOf<AstNode?>()
        kids.add(constraintBody)
        return kids.filterNotNull()
    }

    public override fun <R, C> accept(visitor: AstVisitor<R, C>, ctx: C): R =
        visitor.visitConstraint(this, ctx)
}
